test/client/metadataApiDeploy.test.ts

```diff
@@ -35,6 +35,9 @@ describe('MetadataApiDeploy', () => {
 
   describe('Lifecycle', () => {
     describe('start', () => {
+      it('should not convert zip, but read from fs', async () => {});
+      it('should not mdapiDir, but generate zip buffer from it', async () => {});
```

Comment:
You'll have to modify `stubMetadataDeploy()` in `transferOperations.ts` to stub all the additional fs, zip, and streaming code you added in `getZipBuffer` and return the stubs from `stubMetadataDeploy`.  You'll also have to modify that function to return an `operation` with an instance of `MetadataApiDeploy` created with the mdapiPath and zipPath options.  Your test would then call `operation.start()` like the others, and you would verify stubs were called or not called with the expected args.